httpbackoff.go

```
@@ -102,19 +110,28 @@ func (httpRetryClient *Client) Retry(httpCall func() (resp *http.Response, tempE
 		}
 		// now check if http response code is such that we should retry [500, 600)...
 		if respCode := response.StatusCode; respCode/100 == 5 {
+			// If this is not the last attempt, the request will be retried and
+			// the response body should be closed. If this is the last attempt,
+			// Retry will return an error, so caller isn't responsible for
+			// closing response body. Therefore, in either case, response body
+			// needs to be closed.
+			defer response.Body.Close()
```

I'm not seeing the connection between this comment and this line of code, and neither seems aligned with the header comment.  The code doesn't seem concerned with the distinction of last vs. non-final retry.  And, if this was a 5xx error, then we expected the caller to close the body, right?

Also, `defer` seems to result in ignoring an error here.  Is that intentional?